src/test/java/org/example/spi/InstancioServiceProviderImpl.java

```diff
@@ -0,0 +1,35 @@
+package org.example.spi;
+
+import org.example.AbstractEntityData;
+import org.example.ValidityManager;
+import org.instancio.Random;
+import org.instancio.generator.Generator;
+import org.instancio.spi.InstancioServiceProvider;
+
+public class InstancioServiceProviderImpl implements InstancioServiceProvider {
+
+    @Override
+    public GeneratorProvider getGeneratorProvider() {
+        return (node, generators) -> {
+            if (node.getTargetClass() == ValidityManager.class) {
+                return new ValidityManagerGenerator(node.getParent().getTargetClass());
```

Comment:
The snapshot version adds the `Node,getParent()` method which allows to retrieve the annotation value.